src/redux/reducers/tags.ts

```diff
@@ -0,0 +1,62 @@
+import {
+  createAsyncThunk,
+  createSlice,
+  SliceCaseReducers,
+} from "@reduxjs/toolkit";
+
+import instance from "../../services/api";
+import { LE, Tag } from "../../types";
+import { Pagination } from "../../types/mock-api-types";
+
+export interface TagsStore {
+  tags: LE<Pagination<Tag>>;
+}
+
+const tagsInitialStore: TagsStore = {
+  tags: {
+    page: 1,
+    limit: 10,
+    docs: [],
+    hasNextPage: true,
+  },
+};
+
+const fetchTags = createAsyncThunk<
+  Pagination<Tag>,
+  { query: string } | undefined
+>("tags/fetchTags", async (filters) => {
+  const { query } = filters || {};
+
+  const response = await instance.get("api/tags", {
+    params: {
+      query: { name: { $regex: query, $options: "i" } },
```

Comment:
```suggestion
      query: { name: { $regex: searchString, $options: "i" } },
```

it'll be better name I think